### DISCUSSION TITLE
Syntaxis to write to datapoint 761-767 (MixerOneCircuitTimeSchedule) on Vitodens 100W

### DISCUSSION THREAD

**Discussion by itrankolov** [open3e/open3e#91] [General]
Hello, 

I need to change the shedule that is there by default and I don't know how to cunstruct the write payload. Can you help me a bit?

**Comment by itrankolov** [comment]
Well I think I've managed to to that, now I have another problem. I want to run the boiler based on flow temperature and I don't know how to properly set the flow temperature. At the moment 527 (FlowTemperatureTargetSetpoint) is set at 20.0C, but when I try to update that value it does not change, but 987 (MixerOneCircuitFlowTemperatureTargetSetpoint) changes instead! I cannot set 987 directly, only via 527. On top of that this only works if 1415 (MixerOneCircuitOperationState) is in "reduced" state, which was controlled by the Schedule, which I figured out how to disable. So I'm completely lost .... Please give me some direction/guidance on how to proceed with figuring this out.

**Reply by philippoo66** [reply]
I think you set the boiler to constant temperature mode? in this case utilizing the heating externally demanded option, 
528 : O3EInt16(2, "ExternalTargetFlowTemperatureSetpoint"),
might be the value to modify? I hope this will not get overwritten by the analog input, but perhaps there is anonther setting to avoid that in case...

this should work independently of any heating schedulings etc (perhaps except hot water). for what reason do you need scheduling?

**Reply by Wolfgang-03** [reply]
I only have a Vitocal, but it works in such a way that I cannot set the 987 (MixerOneCircuitFlowTemperatureTargetSetpoint) directly. The Vitocal has the value 424 (MixerOneCircuitRoomTemperatureSetpoint), in which 3 temperatures are stored, for comfort, normal and reduced. Depending on which operating program (1415 MixerOneCircuitOperationState) is active, the value for 1643 (MixerOneCircuitCurrentTemperatureSetpoint) is then derived from one of the 3 values. The value for 987 is then calculated internally by the Vitocal, depending on 274 (OutsideTemperatureSensor) and 1643.

So if I want to change the setpoint for the flow temperature (987), I change the setpoint temperature for the heating circuit (424).

I do this with Home Assistant and it now works quite well for me. I use it to control the flow temperature up to 55 degrees when enough solar power is available. 

Comfort 190 = 19.0 °C
Standard 170 = 17.0 °C
Reduced 150  = 15.0 °C
        
        {"mode":"write-raw","data":[[424,"be00aa009600000000"]]}

Com 260
Sta 240
Red 220
        
        {"mode":"write-raw","data":[[424,"0401f000dc00000000"]]}

**Reply by philippoo66** [reply]
ak ok, I thought regarding the backup gas boiler...

**Reply by itrankolov** [reply]
This does not work for me. Just tried it. No matter what I set in 424 it does not get picked. I also cannot set directly 1415 for operating program, it picks the program from the schedule - that is why I wanted to get rid of it. Now when there is no schedule it is constantly in reduced mode, but it does not care for the settings in 424 and the only way to control flow temp for now is to set 527 which in term sets 987 to whatever I've set 527.

**Reply by Wolfgang-03** [reply]
OK, the Vitodens seems to have a different behavior than the Vitocal. I can't change 1415 on the Vitocal either. Perhaps you can simply set all days and all hours to "Standard" in the time program (ViCare-App), then the system should always run in "Standard mode". 
Does your system not have an outdoor temperature sensor? If I assume that the 527 is for the Vitodes what the 424 is for the Vitocal, then it should work in the same way:

424/527 + 1415 (depending on settings, on my system +/- 2) = 1643 + X (X depending on OutsideTemp) = 987

![image](https://github.com/open3e/open3e/assets/80253488/7edcaf9d-b4f7-4991-90de-26f7f9e840d0)

See yellow marking. With the same specified room temperature (1643), the target flow temperature (987) is different if the outside temperature is different.

**Reply by itrankolov** [reply]
Hello,

I don't have an outdoor sensor and I don't want to have (don't see any benefit of having). You're describing a setup controlled by room temperature, but in mine I want to control only the flow temperature and that is why I want to be able to control it directly. If I try to set 527, it does not change, instead 987 changes. In the vicare app I can change the temp for each mode, but not via CAN. It seems there is another datapoint that is not listed/found yet where this is set. I don't know how all theese datapoints are discovered ... so I can try to discover more myself.
